### PR TITLE
Enhanced enum features for `AnimationStatus`

### PR DESCRIPTION
**Pull request by nate-thegrate**
Based on issue #147799, this pull request adds two `AnimationStatus` getters.

```dart
bool get isRunning => switch (this) {
  forward   || reverse   => true,
  completed || dismissed => false,
};

bool get aimedForward => switch (this) {
  forward || completed => true,
  reverse || dismissed => false,
};
```

I also added a `.toggle()` method for animation controllers that makes use of `aimedForward`.

### CODE REVIEW

File: packages/flutter/test/animation/animation_controller_test.dart [hunk 1]
```diff
@@ -187,6 +187,8 @@ void main() {
     expect(controller.value, moreOrLessEquals(0.0));
     controller.stop();
 
+    controller.dispose();
+
```
Comment (nate-thegrate) [inline packages/flutter/test/animation/animation_controller_test.dart:191]: I noticed that in another test, the controller was being reassigned without a `.dispose()` call, so I added this as well.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 2]
```diff
@@ -527,6 +527,45 @@ class AnimationController extends Animation<double>
...
+  /// Toggles the direction of this animation, based on whether it's [aimedForward].
+  ///
+  /// Specifically, this function acts the same way as [reverse] if the [status] is
+  /// either [AnimationStatus.forward] or [AnimationStatus.completed], and acts as
+  /// [forward] for [AnimationStatus.reverse] or [AnimationStatus.dismissed].
+  ///
+  /// The most recently returned [TickerFuture], if any, is marked as having been
+  /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
+  /// derivative future completes with a [TickerCanceled] error.
+  TickerFuture toggle({ double? from }) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:539]: Can you document the parameter?

File: packages/flutter/lib/src/animation/animation.dart [hunk 3]
```diff
@@ -27,7 +27,29 @@ enum AnimationStatus {
...
+  completed;
+
+  /// Whether the animation is stopped at the beginning.
+  bool get isDismissed => this == dismissed;
+
+  /// Whether the animation is stopped at the end.
+  bool get isCompleted => this == completed;
+
+  /// Whether the animation is running in either direction.
+  bool get running => switch (this) {
```
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/animation/animation.dart]: Sounds good to me!

File: packages/flutter/lib/src/animation/animation.dart [hunk 4]
```diff
@@ -27,7 +27,29 @@ enum AnimationStatus {
...
+    forward   || reverse   => true,
+    completed || dismissed => false,
+  };
+
+  /// {@template flutter.animation.aimedForward}
+  /// Whether the current aim of the animation is toward completion.
+  ///
+  /// Useful for toggling animations back and forth.
+  /// {@endtemplate}
+  bool get aimedForward => switch (this) {
```
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/animation/animation.dart:50]: me too 🥲

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 2 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/animation/animation_controller.dart:548]: Sure can!

I copied the documentation over from the `forward`/`reverse` functions, so I'll document the parameter for those as well.

File: packages/flutter/lib/src/animation/animation.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/src/animation/animation.dart:51]: I asked gemini, it came up with the very explicit `isForwardOrCompleted` name. Still not great, I like it slightly better then aimedForward, though.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/animation/animation.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/animation/animation.dart:51]: I'm good with that!

**Pull request merged**